src/main/java/duke/save/SaveTaskListToText.java

```diff
@@ -0,0 +1,346 @@
+
+                numberOfTasksAdded += 1;
+            } else if (commandHandle.isDeadline()) {
+                int done = Integer.parseInt(currentCommand.split("-/-")[1]);
+                String taskName = currentCommand.split("-/-")[2];
+                String by = currentCommand.split("-/-")[3];
+
+                tasks.add(new Deadline(taskName, by));
+
+                if (done == 1) {
+                    tasks.get(numberOfTasksAdded).markAsDone();
+                }
+
+                numberOfTasksAdded += 1;
+            }
+        }
+
+        scanText.close();
+    }
+
+    /**
+     * Checks if directory exist first. If it doesn't, it means the duke.txt is also missing.
+     * In such a case, make directory and then create duke.txt file there. Then if directory
+     * exist, but duke.txt is missing, add duke.txt.
+     *
+     * @param tasks
+     * @return
+     */
+    public static int loadTask(ArrayList<Task> tasks) throws IOException {
+        if (directoryOfTaskText.exists() == false) {
+            directoryOfTaskText.mkdir();
+            dukeTaskText.createNewFile();
+        } else if (dukeTaskText.exists() == false) {
+            dukeTaskText.createNewFile();
+        } else {
+            fileTaskCopy(tasks);
+        }
+
+        return numberOfTasksAdded;
+    }
+
+    // Solution below adapted from
+    // https://stackoverflow.com/questions/5600422/method-to-find-string-inside-of-the-text-file-then-getting-the-following-lines
+    public static boolean checkLineExist(String stringToCheck) {
+        File file = new File("./data/duke.txt");
+
+        try {
+            Scanner scanner = new Scanner(file);
+
+            int lineNum = 0;
+
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                lineNum++;
+
+                if (line.equals(stringToCheck)) {
+                    scanner.close(); // Don't forget!
+                    return true;
+                }
+            }
+            scanner.close();
+        } catch(FileNotFoundException e) {
+            System.out.println("File not found!");
+        }
+
+        return false;
+    }
+
+    // Solution below adapted from https://stackoverflow.com/questions/1625234/how-to-append-text-to-an-existing-file-in-java
+    public static boolean saveToDo(String taskName) {
+        String textToWrite1 = "t-/-0-/-" + taskName;
+        String textToWrite2 = "t-/-1-/-" + taskName;
+
+        try {
+            if (checkLineExist(textToWrite1) == false && checkLineExist(textToWrite2) == false) {
+                String dukeDirectory = "data/duke.txt";
+
+                File file = new File(dukeDirectory);
+                boolean fileIsEmpty = false;
+
+                if (file.length() == 0) {
+                    fileIsEmpty = true;
+                }
+
+                FileWriter addToDo = new FileWriter(dukeDirectory, true); //the true will append the new data
+                if (fileIsEmpty == false) {
```

Comment:
This boolean condition can be written as `!fileIsEmpty` to make it less verbose.